sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -1112,4 +1112,29 @@ class SparkSqlAstBuilder extends AstBuilder {
       withIdentClause(ctx.identifierReference(), UnresolvedNamespace(_)),
       cleanedProperties)
   }
+
+  /**
+   * Create a [[ShowCollationsCommand]] command.
+   * Expected format:
+   * {{{
+   *   SHOW identifier? COLLATIONS ((FROM | IN) ns=identifierReference)? (LIKE? pattern=stringLit);
+   * }}}
+   */
+  override def visitShowCollations(ctx: ShowCollationsContext): LogicalPlan = withOrigin(ctx) {
+    val ns = if (ctx.ns != null) {
+      withIdentClause(ctx.ns, UnresolvedNamespace(_))
+    } else {
+      CurrentNamespace
+    }
+    val (userScope, systemScope) = Option(ctx.identifier)
+      .map(_.getText.toLowerCase(Locale.ROOT)) match {
+      case None | Some("all") => (true, true)
+      case Some("system") => (false, true)
+      case Some("user") =>
+        throw QueryParsingErrors.showCollationsUnsupportedError("user", ctx.identifier())
```

Comment:
Could we maybe remove this scoping totally and just return the current namespace? I would argue for now it is not necessary to allow any additional feature that we do not actually need or have implemented. So simple SHOW COLLATIONS (LIKE pattern)? should be enough and we can just keep ShowCollationsCommand to use CurrentNamespace, userScope=false, systemScope=true. What do you think about that? This way I would say we are keeping the idea of what we will do when user defined collations are in, but will not pollute users with unnecessary errors.

Reply:
Which of the following is more in line with what you want to think?

- Option 1
```scala
/**
   * Create a [[ShowCollationsCommand]] command.
   * Expected format:
   * {{{
   *   SHOW COLLATIONS ((FROM | IN) ns=identifierReference)? (LIKE? pattern=stringLit)?;
   * }}}
   */
  override def visitShowCollations(ctx: ShowCollationsContext): LogicalPlan = withOrigin(ctx) {
    val ns = if (ctx.ns != null) {
      withIdentClause(ctx.ns, UnresolvedNamespace(_))
    } else {
      CurrentNamespace
    }
    val pattern = Option(ctx.pattern).map(x => string(visitStringLit(x)))
    ShowCollationsCommand(ns, userScope = false, systemScope = true, pattern)
  }
```

- Option 2
```scala
/**
   * Create a [[ShowCollationsCommand]] command.
   * Expected format:
   * {{{
   *   SHOW COLLATIONS (LIKE? pattern=stringLit)?;
   * }}}
   */
  override def visitShowCollations(ctx: ShowCollationsContext): LogicalPlan = withOrigin(ctx) {
    val ns = if (ctx.ns != null) {
      withIdentClause(ctx.ns, UnresolvedNamespace(_))
    } else {
      CurrentNamespace
    }
    val pattern = Option(ctx.pattern).map(x => string(visitStringLit(x)))
    ShowCollationsCommand(ns, pattern)
  }
```

- Option 3
```scala
/**
   * Create a [[ShowCollationsCommand]] command.
   * Expected format:
   * {{{
   *   SHOW COLLATIONS (LIKE? pattern=stringLit)?;
   * }}}
   */
  override def visitShowCollations(ctx: ShowCollationsContext): LogicalPlan = withOrigin(ctx) {
    val pattern = Option(ctx.pattern).map(x => string(visitStringLit(x)))
    ShowCollationsCommand(pattern)
  }
```

Reply:
Option 3 seems the best to me for now. Option 2 will not work, as you do not have ns in context. Option one is good, but we add additional computation of UnresolvedNamespace even though we will never use it later, until user defined collations are in. What I suggested in the beginning is to explore Show Functions as it gave a perspective of things we need to keep in mind, but I would say Option 3 is the best in line with features we currently have.

Reply:
Okay

Reply:
I wrote `Option 2` wrong, I updated it.

Reply:
Yeah, now Option 2 seems fine as well if you ask me. It is up to you to decide.